contrib/platform/src/com/sun/jna/platform/win32/WinCrypt.java

```diff
@@ -129,10 +129,12 @@ public DATA_BLOB(String s) {
             if (cAttribute == 0) {
                 return new CRYPT_ATTRIBUTE[0];
             } else {
-                return (CRYPT_ATTRIBUTE[]) Structure.newInstance(
-                        CERT_EXTENSION.class,
+                CRYPT_ATTRIBUTE[] result = (CRYPT_ATTRIBUTE[]) Structure.newInstance(
+                        CRYPT_ATTRIBUTE.class,
                         rgAttribute)
                         .toArray(cAttribute);
+                result[0].read();
```

Comment:
Does this read the whole array or just element 0? (Same question repeatedly.)

Reply:
The elements 1-X are initialized by `toArray`. The first element of the new array is skipped for initialization in `Structure#toArray` and thus needs to be explicitly read. (Applies to all places where this pattern can be found).